tests/integration/suite/daprd/serviceinvocation/http/sserelay.go

```diff
@@ -0,0 +1,320 @@
+					log.Printf(">>SRVC>> recv %d bytes: %q\n", len(line), line)
+					s.eventsC.Store(count, line)
+					count++
+					// Forward exactly what we received (includes \n)
+					io.WriteString(w, line)
+
+					// Flush immediately for real-time streaming
+					flusher.Flush()
+				}
+			}
+		}),
+	)
+
+	s.daprdA = daprd.New(t, daprd.WithAppID("daprdA"), daprd.WithAppPort(s.srvA.Port()))
+	s.daprdB = daprd.New(t, daprd.WithAppID("daprdB"), daprd.WithAppPort(s.srvB.Port()))
+	s.daprdC = daprd.New(t, daprd.WithAppID("daprdC"), daprd.WithAppPort(s.srvC.Port()))
+
+	return []framework.Option{
+		framework.WithProcesses(s.srvA, s.srvB, s.srvC, s.daprdA, s.daprdB, s.daprdC),
+	}
+}
+
+func (s *sserelay) Run(t *testing.T, ctx context.Context) {
+	s.daprdA.WaitUntilRunning(t, ctx)
+	s.daprdB.WaitUntilRunning(t, ctx)
+
+	client := client.HTTPWithTimeout(t, 60*time.Second)
+
+	url := fmt.Sprintf("http://localhost:%d/TOC", s.srvC.Port())
+	req, err := http.NewRequest(http.MethodGet, url, nil)
+	require.NoError(t, err)
+	req.Header.Set("Accept", "text/event-stream")
+	req.Header.Set("Cache-Control", "no-cache")
+	req.Header.Set("Accept-Encoding", "identity")
+
+	resp, err := client.Do(req)
+	if resp != nil {
+		defer func() {
+			cErr := resp.Body.Close()
+			if cErr != nil {
+				t.Log("Response body close error: ", cErr)
+			}
+		}()
+	}
+	require.NoError(t, err)
+
+	log.Printf("status=%s proto=%s", resp.Status, resp.Proto)
+
+	// Verify SSE headers
+	require.Equal(t, "text/event-stream", resp.Header.Get("Content-Type"))
+
+	// And modify the test to be more resilient to connection issues:
+	// Replace the current event counting with timeout-based reading
+	timeoutCtx, timeoutCancel := context.WithTimeout(t.Context(), 10*time.Second)
+	defer timeoutCancel()
+
+	// Use direct reader for real-time streaming
+	done := make(chan struct{})
+	errChan := make(chan error)
+	eventCount := atomic.Int32{}
+	go func() {
+		defer close(done)
+		reader := bufio.NewReader(resp.Body)
+		for {
+			line, rErr := reader.ReadString('\n')
+			if rErr == io.EOF {
+				return
+			}
+			if rErr != nil {
+				if !errors.Is(rErr, context.Canceled) {
+					errChan <- rErr
+				}
+				return
+			}
+
+			log.Printf(">>TEST>> recv %d bytes: %q\n", len(line), line)
+
+			val, ok := s.eventsA.Load(int(eventCount.Load()))
+			assert.True(t, ok)
+			assert.Equal(t, val, line)
+
+			val, ok = s.eventsB.Load(int(eventCount.Load()))
+			assert.True(t, ok)
+			assert.Equal(t, val, line)
+
+			val, ok = s.eventsC.Load(int(eventCount.Load()))
+			assert.True(t, ok)
+			assert.Equal(t, val, line)
+
+			eventCount.Add(1)
+			// Stop after receiving a certain number of events for testing
+			if eventCount.Load() > 5 {
+				errChan <- errors.New("received too many events")
+			}
+		}
+	}()
+
+	select {
+	case <-done:
+		assert.EqualValues(t, 5, eventCount.Load())
+		log.Printf("done; received %d events", eventCount.Load())
+	case err = <-errChan:
+		require.NoError(t, err)
+	case <-timeoutCtx.Done():
```

Comment:
```suggestion
	case <-time.After(10 * time.Second)
```
This way you don't need to create a context for this